File: sdk/client/java/openapi-test/src/test/java/opensdk/sdk/apis/txpool/TxpoolInspectTest.java
package opensdk.sdk.apis.txpool;

import opensdk.sdk.apis.constant.UrlConstants;
import opensdk.sdk.models.TxpoolInspectResponse;
import org.junit.jupiter.api.DisplayName;
import org.junit.jupiter.api.Test;
import org.klaytn.OpenSDK;

import java.io.IOException;

@DisplayName("Txpool RPC Test")
public class TxpoolInspectTest {
    private final OpenSDK sdk = new OpenSDK(UrlConstants.SERVER_URL);

    @Test
    @DisplayName("RPC txpool_inspect")
    void whenRequestValid_ThenCall200ResponseReturns() throws IOException {
        TxpoolInspectResponse response = sdk.txpool.inspect().send();
        response.getResult();
    }
}

File: web3rpc/sdk/client/javascript/openapi-test/test/txpool/TxpoolInspect.test.js
const OpenSdk = require("@klaytn/web3rpc");
const { expect } = require("@jest/globals");
const { RPC } = require("../constant");


const sdk = new OpenSdk.TxpoolApi(new OpenSdk.ApiClient(RPC));

describe('txpool_inspect API', () => {
    test('should return txpool_inspect', (done) => {

        let callbackOne = function (error, data, response) {
            expect(error).toBeNull();
            expect(data).toBeDefined()
            expect(data).toHaveProperty('pending' || 'queued')
            done();
        };

        sdk.inspect({}, callbackOne);
    });
});

